pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1273,6 +1273,17 @@ protected void handleProducer(final CommandProducer cmdProducer) {
                             if (!Strings.isNullOrEmpty(initialSubscriptionName)
                                     && topic.isPersistent()
                                     && !topic.getSubscriptions().containsKey(initialSubscriptionName)) {
+                                if (!this.getBrokerService().isAllowAutoSubscriptionCreation(topicName)) {
+                                    String msg =
+                                            "Could not create the initial subscription due to the auto subscription "
+                                                    + "creation is not allowed.";
```

Comment:
@RobertIndie - is it documented that the DLQ producer will fail to create in this case?

Reply:
@RobertIndie Please also update the document.

Reply:
Thanks. I have updated the doc. PTAL @michaeljmarshall @codelipenghui